app/src/main/java/com/x8bit/bitwarden/ui/platform/base/util/EventsEffect.kt

```diff
@@ -12,11 +15,23 @@ import kotlinx.coroutines.flow.onEach
 @Composable
 fun <E> EventsEffect(
     viewModel: BaseViewModel<*, E, *>,
+    lifecycleOwner: Lifecycle = LocalLifecycleOwner.current.lifecycle,
     handler: suspend (E) -> Unit,
 ) {
     LaunchedEffect(key1 = Unit) {
         viewModel.eventFlow
+            .filter {
+                it is BackgroundEvent ||
+                    lifecycleOwner.currentState.isAtLeast(Lifecycle.State.RESUMED)
+            }
             .onEach { handler.invoke(it) }
             .launchIn(this)
     }
 }
+
+/**
+ * Almost all the events in the app involve navigation or toasts. To prevent accidentally
+ * navigating to the same view twice, by default, events are ignored if the view is not currently
+ * resumed. To avoid that restriction, specific events can conform to [BackgroundEvent].
+ */
+interface BackgroundEvent
```

Comment:
Would it make more sense for this to live in its own file maybe? I'm just not sure we have other examples like this where an interface is defined in the same file as a composable function, but maybe I'm wrong about that.

Reply:
Sure, we can do that! I only put it here because I didn't know if there was a precedent for creating a file with only one line of code in it